Automattic/jetpack
# Update/my jetpack status handling

## Proposed changes:

* This diff updates how the active and inactive statuses are handled for products in My Jetpack
* More granular checks for paid plans and free offerings are introduced
* `has_required_plan` function has been removed in favor of `has_paid_plan_for_product`, `has_free_plan_for_product` and `has_any_plan_for_product`
* Separate connection statuses are established for site and user connections

### Other information:


## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






* Using the Jetpack Beta plugin, apply this PR on a brand new Jetpack site that has the main Jetpack plugin.
* Before connecting the site, visit My Jetpack. Confirm that all the product cards are showing as "inactive". This should be true since the site has never been connected before.
![Screenshot 2024-05-14 at 5 53 26 PM](https://github.com/Automattic/jetpack/assets/18016357/eadac7e4-f459-4d74-ba0c-f658ed05ddde)

* Now, connect the site
* Return to My Jetpack. Now, any feature that is contained within the main Jetpack plugin (Creator, Stats) will show as "Active". AI will show as "Needs Sign in". Akismet may be active as well depending on your test site and if an API key has been added for Akismet.
![Screenshot 2024-05-14 at 5 54 11 PM](https://github.com/Automattic/jetpack/assets/18016357/16bbf0f0-bccc-44f4-a12d-680dcc7dab15)

* If you connect your user account, the AI product should show as "Active" as well.
![Screenshot 2024-05-14 at 5 54 50 PM](https://github.com/Automattic/jetpack/assets/18016357/78536c1b-fbf6-4dbe-8317-5fd4182e94a4)

* If you disconnect Jetpack, all products that were active before will now show "Needs Connection"
![Screenshot 2024-05-14 at 5 55 56 PM](https://github.com/Automattic/jetpack/assets/18016357/eb6e98cb-2a58-4785-b365-cd2670cc358d)

* Test purchasing and activating other products to make sure that the active/inactive statuses make sense

-----

* Now, spin up a new test site, but this time, only install the Jetpack Social Standalone plugin (and the beta testing plugin of course)
* With this PR applied via the Beta Plugin, Jetpack social should show as inactive before the site is connected
![Screenshot 2024-05-15 at 3 11 06 PM](https://github.com/Automattic/jetpack/assets/18016357/406029be-f7d5-4ef1-8e32-98d48ec3f170)
* If you connect the site but not the user, Social should show as "needs sign-in"
![Screenshot 2024-05-15 at 3 11 26 PM](https://github.com/Automattic/jetpack/assets/18016357/b6bb67a4-d8e2-4e7e-9ae8-6bf5a9c45216)
* After the site is connected, Jetpack social should show as "active"
![Screenshot 2024-05-15 at 3 12 00 PM](https://github.com/Automattic/jetpack/assets/18016357/1016b9d7-3d8b-490e-9187-a519c6ebd929)
